### ISSUE TITLE
SteamCMD missing file permissions that it has 

### ISSUE THREAD

**Issue by mja00** [ValveSoftware/steam-for-linux#10979]
#### Your system information

* Distribution (e.g. Ubuntu): Ubuntu
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes
* Steam Logs: Ephemeral container
* GPU: N/A

#### Please describe your issue in as much detail as possible:
When attempting to install a server through SteamCMD it fails with a `ERROR! Failed to install app '105600' (Missing file permissions)` error. I've check file permissions and SteamCMD has them. This worked perfectly fine yesterday but now no longer works.

#### Steps for reproducing this issue:

1. Install a server with SteamCMD
2. See it error out

**Comment by Lord-Leonard** [comment]
getting this as well, any info?

**Comment by mja00** [comment]
I ended up having to set a `HOME` env var to a folder it could write to. SteamCMD was attempting to write to the `/` folder which caused it to do `//Steam/logs` as its logs directory and failing.

**Comment by mja00** [comment]
It seems in some recent update whatever code that writes the logs now fatally errors out if it can't actually write them.

**Comment by nevesm** [comment]
I am facing the same issue:

`ERROR! Failed to install app '730' (Missing file permissions)`

I dont get it how you solved it

**Comment by mja00** [comment]
It's possible you're having a different bug. When SteamCmd first runs it'll say something about redirecting errors to a directory. If that directory is one the running user cannot write to then it'll fail with missing permissions. I've got no clue why but it does.

**Comment by nevesm** [comment]
Oh i see, i think mine is `/root/Steam/logs`
![image](https://github.com/ValveSoftware/steam-for-linux/assets/44535389/654793f6-5aab-4b27-981f-1ec39ff6e944)

**Comment by mja00** [comment]
Yeah it reads from the HOME environment variable to set the directory so it is changeable.

**Comment by nevesm** [comment]
I'll try to give the necessary permissions to this folder, 1 sec

**Comment by nevesm** [comment]
Not working yet, i think you should keep this issue open because this can crash other servers not only ours

**Comment by dudleycodes** [comment]
Setting the `HOME` environment variable solved my problem using [LacledesLAN/SteamCMD](https://github.com/LacledesLAN/SteamCMD).

**Comment by Lord-Leonard** [comment]
For me, it was actualy file permissions .. 
I dont quiet know what it was, but after executing these commands it started working again. 

cd ~/steamcmd

# Ensure correct permissions for arma3 directory
chmod -R u+rwX ~/steamcmd/arma3

# Ensure correct ownership for arma3 directory
chown -R steam:steam ~/steamcmd/arma3

# Run SteamCMD as the steam user
sudo -u steam ~/steamcmd/steamcmd.sh

# Inside SteamCMD, set the installation directory and update Arma 3
force_install_dir ~/steamcmd/arma3
login username
app_update 233780

Of cause, ensure your paths match.

**Comment by Ric-F** [comment]
In my case for a Project Zomboid server running https://github.com/Danixu/project-zomboid-server-docker , to eliminate the file permissions problem on start I needed to update my Dockerfile like so:

```
ENV HOME "${HOMEDIR}"
ENV STEAMAPPID 380870
ENV STEAMAPP pz
ENV STEAMAPPDIR "${HOMEDIR}/${STEAMAPP}-dedicated"

```

The only line added was the top one where I re-used the HOMEDIR environment variable for HOME.
